src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/GoGoDuck.java

```diff
@@ -409,25 +405,37 @@ public static int execute(List<String> command) throws Exception {
         builder.redirectErrorStream(true);
 
         final Process process = builder.start();
+        StringBuilder bashLog = new StringBuilder();
 
         try (
             InputStream is = process.getInputStream();
             InputStreamReader isr = new InputStreamReader(is);
             BufferedReader br = new BufferedReader(isr)
         ) {
             String line;
+
             while ((line = br.readLine()) != null) {
                 logger.info(line);
+                bashLog.append(line);
             }
 
             process.waitFor();
 
+            if (process.exitValue() != 0) {
```

Comment:
`processListener.isCancelled(true) ` worked well for me. I believed you were going to expand on this in other functions. As @anguss00 commented previously  `to consider setting the processListener.isCancelled(true) whenever we throw a GGD exception`